python/python/lance/vector.py

```diff
@@ -5,30 +5,14 @@
 
 from __future__ import annotations
 
-import logging
-import re
-import tempfile
-from typing import TYPE_CHECKING, Any, Iterable, List, Literal, Optional, Union
+from typing import Optional, Union
 
 import pyarrow as pa
-from tqdm.auto import tqdm
 
-from . import write_dataset
-from .cuvs.kmeans import KMeans as KMeansCuVS
 from .dependencies import (
-    _CAGRA_AVAILABLE,
-    _RAFT_COMMON_AVAILABLE,
     _check_for_numpy,
-    torch,
```

Comment:
I thought this was ok to have, since it's lazily used and won't import the real thing in an env without torch installed? 
The intention of all the code in vector.py was to only use torch-based code after importing torch inside a particular function

Reply:
Yeah I would have thought so too 🤔

Reply:
After discussing with Weston: Seems like there's some recently introduced accidental import chain all the way back to the torch-based distance comparisons. One of us will probably find a simpler fix tomorrow.
This PR isn't a bad refactor though, so might be worth landing at some point.